### PR TITLE
Added support for using WorkloadIdentity Auth

### PR DESCRIPTION
**Pull request by ociappara**
Allow users to auth with WorkloadIdentityCredential

### CODE REVIEW

**Comment by ociappara** [comment]
Still working on testing end to end code changes

File: assemblyline_ui/api/v4/authentication.py [hunk 1]
```diff
@@ -661,7 +676,16 @@ def oauth_validate(**_):
                 oauth_provider_config = config.auth.oauth.providers[oauth_provider]
 
                 # Validate the token
-                if oauth_provider_config.validate_token_with_secret or oauth_provider_config.app_provider:
+                if oauth_provider_config.auto_no_secret:
+
+                    workload_credential = WorkloadIdentityCredential(
+                        additionally_allowed_tenants=[oauth_provider_config.client_tenent])
```
Comment (cccs-rs) [inline assemblyline_ui/api/v4/authentication.py:682]: oauth_provider_config.client_tenent should already be a list of strings, haven't looked into the class but I assume it doesn't require a `List[List[str]]`?

File: assemblyline_ui/api/v4/authentication.py [hunk 2]
```diff
@@ -661,7 +676,16 @@ def oauth_validate(**_):
...
+
+                    token = workload_credential.get_token(
+                            oauth_provider_config.client_scope
+                        , tenant_id=oauth_provider_config.client_tenent)
+
+                if (oauth_provider_config.validate_token_with_secret or oauth_provider_config.app_provider) and not token:
```
Comment (cccs-sgaron) [inline assemblyline_ui/api/v4/authentication.py:688]: this should just be an else if and you would not have to add the "and not token"

**Review by cccs-sgaron — CHANGES_REQUESTED** [review]

File: assemblyline_ui/api/v4/authentication.py [hunk 3]
```diff
@@ -661,7 +676,16 @@ def oauth_validate(**_):
...
                 # Validate the token
-                if oauth_provider_config.validate_token_with_secret or oauth_provider_config.app_provider:
+                if oauth_provider_config.auto_no_secret:
+
+                    workload_credential = WorkloadIdentityCredential(
+                        additionally_allowed_tenants=[oauth_provider_config.client_tenent])
+
+                    token = workload_credential.get_token(
+                            oauth_provider_config.client_scope
+                        , tenant_id=oauth_provider_config.client_tenent)
```
Comment (cccs-sgaron) [inline assemblyline_ui/api/v4/authentication.py:686]: Same typo as the base repo. **client_tenent** => **client_tenants**

File: assemblyline_ui/api/v4/authentication.py (hunk 1 above)
Comment (C-Oliver) [reply assemblyline_ui/api/v4/authentication.py]: Done.

File: assemblyline_ui/api/v4/authentication.py (hunk 2 above)
Comment (C-Oliver) [reply assemblyline_ui/api/v4/authentication.py]: Done.

**Comment by ociappara** [comment]
Are we okay to merge this to Dev ?

**Comment by cccs-sgaron** [comment]
Merging it into dev as: https://github.com/CybercentreCanada/assemblyline-ui/pull/978

**Review by cccs-sgaron — APPROVED** [review]

**Pull request merged**